articles/150_message_field_names.md

```diff
@@ -0,0 +1,94 @@
+---
+layout: default
+title: Message Field Name Conflicts
+permalink: articles/message_field_conflicts.html
+abstract:
+  As we move to support more platforms there are we are running into predefined fields and constants that conflict with languages or implementation keywords and macros.
+  This defines how to deal with existing and potential future conflicts.
+published: true
+author: '[Tully Foote](https://github.com/tfoote)'
+categories: Interfaces
+---
+
+{:toc}
+
+# {{ page.title }}
+
+<div class="abstract" markdown="1">
+{{ page.abstract }}
+</div>
+
+Original Author: {{ page.author }}
+
+## Scope
+
+This article specifies how we will deal with conflicting fields and constants for different languages.
+In the use of the IDL (TODO (tfoote) reference) there will inevitably be conflicting symbols, keywords, or definitions.
+
+## Background and Motivation
+
+This has come up specifically as we're adding support for Windows for ROS2.
+`winnt.h` defines several macros that conflict with existing enumerations. 
+It is also expected to happen as support for new languages are added.
+We cannot know all the potential future keywords and restrictions from a language which is selected to add support for in the future so we must have a generic solution which will allow future languages to be added without disrupting the existing usages.
+
+## New constants
+
+To avoid conflicts generally all constants should be prefixed with a `k`. This will avoid almost all conflicts.
+
+Google Style Guide: https://google.github.io/styleguide/cppguide.html#Constant_Names
+
+TODO(tfoote) Reference ROS2 style guide
+
+## Language Generator Requirements
+
+In the case that a new language is added and there is a detected existing symbol, the language specific generator is expected to implement a basic deconfliction mangling of the symbol to avoid the conflict.
+
+Each generator will define the mangling procedure and provide a list of symbols for which the procedure will be applied.
+This list should be all known keywords or otherwise conflicting symbols.
```

Comment:
As the parser will not have knowledge of the generator's specific mangling, it may be worth mentioning that each generator will need to ensure that each field is mangled (if needed) and to check that the resulting fields don't collide wit each other (e.g. someone defining a field `class` and a field `class_` in the same msg)

Reply:
I added more specific language to say that this sort of accidental collision should be made not possible.